numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -398,6 +398,9 @@ _strided_to_strided_contig_align_wrap(char *dst, npy_intp dst_stride,
     npy_bool init_dest = d->init_dest, out_needs_api = d->out_needs_api;
 
     for(;;) {
+        if (out_needs_api && PyErr_Occurred()) {
```

Comment:
```suggestion
        /* The caller does not know if a previous call resulted in a python exception.
         * Much of the Python API is unsafe while an exception is in flight, so just
         * skip all the work. Someone higher in the call stack will check for errors
         * and propagate them
         */
        if (out_needs_api && PyErr_Occurred()) {
``
Reword as desired.

Reply:
80 column formatting...

Reply:
github needs a ruler in their editor....